Signal/src/textsecure/Messages/TSMessagesManager.m

```diff
@@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
 }
 
 - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
+    if(content.group!= nil )  {
+        __block BOOL ignoreMessage = NO;
+        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
+            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
+            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
+            if(gThread==nil) {
+                ignoreMessage = YES;
+            }
+        }];
+        if(ignoreMessage) {
+            DDLogDebug(@"recevied message from group that I left, ignoring");
```

Comment:
according to my understanding take away from today the server will continue
to send messages not acked by clients. so if the client doesn't ack these
sorts of messages the right behavior will result.

On Fri, Jan 2, 2015 at 9:03 PM, Christine Corbett Moran <
christine.corbett@gmail.com> wrote:

> this isn't anything to do with the server. the client doesn't send an ack.
>
> On Fri, Jan 2, 2015 at 9:02 PM, Frederic Jacobs <notifications@github.com>
> wrote:
>
>> In Signal/src/textsecure/Messages/TSMessagesManager.m
>> <https://github.com/WhisperSystems/Signal-iOS/pull/248#discussion-diff-22422735>
>> :
>>
>> > @@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
>> >  }
>> >
>> >  - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
>> > +    if(content.group!= nil )  {
>> > +        __block BOOL ignoreMessage = NO;
>> > +        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
>> > +            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
>> > +            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
>> > +            if(gThread==nil) {
>> > +                ignoreMessage = YES;
>> > +            }
>> > +        }];
>> > +        if(ignoreMessage) {
>> > +            DDLogDebug(@"recevied message from group that I left, ignoring");
>>
>> @corbett <https://github.com/corbett> I would strongly advocate against
>> that server-side option.
>>
>> The server IS and SHOULD stay dumb. If we're letting the server know in
>> any way that you left a group, that would be information leakage. Those
>> protocol buffers are encrypted and the server doesn't know it's a group
>> message. So that is not even technically feasible.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/WhisperSystems/Signal-iOS/pull/248/files#r22422735>.
>>
>
>
>
> --
> Dr. Corbett Moran
> Rockets @ http://www.spacex.com
> Physics @ http://www.ICS.uzh.ch  <http://www.ICS.uzh.ch>
> Dev @ http://circleof6app.com
> Dev @ https://whispersystems.org
> www.christinecorbettmoran.com
>
>


-- 
Dr. Corbett Moran
Rockets @ http://www.spacex.com
Physics @ http://www.ICS.uzh.ch  <http://www.ICS.uzh.ch>
Dev @ http://circleof6app.com
Dev @ https://whispersystems.org
www.christinecorbettmoran.com